code/game/turfs/turf.dm

```diff
@@ -191,16 +191,22 @@
 	// First, make sure it can leave its square
 	if(isturf(mover.loc))
 		// Nothing but border objects stop you from leaving a tile, only one loop is needed
-		for(var/obj/obstacle in mover.loc)
-			if(!obstacle.CheckExit(mover, src) && obstacle != mover && obstacle != forget)
+		// This as anything looks odd, since we check istype shortly after, but it's so we can save an istype check for items, which are far more common than other objects.
+		for(var/obj/obstacle as anything in mover.loc)
+			if(isitem(obstacle) || !istype(obstacle) || obstacle == mover || obstacle == forget)
+				continue
+			if(!obstacle.CheckExit(mover, src))
 				mover.Bump(obstacle, TRUE)
 				return FALSE
 
 	var/list/large_dense = list()
-	//Next, check objects to block entry that are on the border
-	for(var/atom/movable/border_obstacle in src)
+	// Next, check for border obstacles on this turf
+	// Everyting inside a turf is an atom/movable.
+	for(var/atom/movable/border_obstacle as anything in src)
+		if(isitem(border_obstacle) || border_obstacle == forget || isnull(border_obstacle))
```

Comment:
Same here, feels questionable

Reply:
`isitem()` is more specific than `atom/movable`